chalicelib/checks/wfr_checks.py

```diff
@@ -1,1449 +1,1436 @@
+            ] = cases_without_sample_processing
+    if sample_processings:
+        sample_processings_found = []
+        sample_processings = format_kwarg_list(sample_processings)
+        found, not_found = validate_items_existence(sample_processings, connection)
+        for sample_processing in found:
+            sample_processings_found.append(sample_processing.get("uuid"))
+        sample_processings_for_meta_workflow += sample_processings_found
+        msg = "%s SampleProcessing(s) were found" % len(sample_processings_found)
+        check.brief_output.append(msg)
+        check.full_output["sample_processings_found"] = sample_processings_found
+        if not_found:
+            msg = "%s SampleProcessing(s) not found" % len(not_found)
+            check.brief_output.append(msg)
+            check.full_output["sample_processings_not_found"] = not_found
+    sample_processings_for_meta_workflow = list(
+        set(sample_processings_for_meta_workflow)
+    )
+    msg = "%s SampleProcessing(s) found to use for MetaWorkflowRun creation" % len(
+        sample_processings_for_meta_workflow
+    )
+    check.brief_output.append(msg)
+    check.full_output[
+        "sample_processing_for_meta_workflow"
+    ] = sample_processings_for_meta_workflow
+    if sample_processings_for_meta_workflow and meta_workflow:
+        msg = "Action will create %s MetaWorkflowRun(s) for MetaWorkflow %s" % (
+            len(sample_processings_for_meta_workflow),
+            meta_workflow,
+        )
+        check.brief_output.append(msg)
+        check.summary = msg
     else:
-        check.summary = 'All Good!'
+        msg = "Could not find information required to create MetaWorkflowRuns"
+        check.brief_output.append(msg)
+        check.summary = msg
+        check.allow_action = False
     return check
 
 
 @action_function()
-def long_running_wfrs_start(connection, **kwargs):
-    """Start runs by sending compiled input_json to run_workflow endpoint"""
-    action = ActionResult(connection, 'long_running_wfrs_start')
-    my_auth = connection.ff_keys
-    long_running_wfrs_check_result = action.get_associated_check_result(kwargs).get('full_output', {})
-    deleted_wfrs = 0
-    status_protected = 0
-    for a_wfr in long_running_wfrs_check_result:
-        # don't deleted if item is in protected statuses
-        if a_wfr['wfr_status'] in ['shared', 'current']:
-            status_protected += 1
-        else:
-            deleted_wfrs += 1
-            for an_item_to_delete in a_wfr['items_to_delete']:
-                ff_utils.patch_metadata({'status': 'deleted'}, an_item_to_delete, my_auth)
-    msg = '{} wfrs were removed'.format(str(deleted_wfrs))
-    if status_protected:
-        msg += ', {} wfrs were skipped due to protected item status.'.format(str(status_protected))
-    action.output = msg
-    action.status = 'DONE'
+def create_meta_workflow_runs_for_items(connection, **kwargs):
```

Comment:
So in the future we add another `check_function` for picking up `PipelineConfig` and implement the runner in this action?

Reply:
Probably, something like that. I believe we'll have default config items for the portal such that we can find them via a search on some standardized properties. It'll depend on how we structure things; perhaps the default config will be linkedTo the MWF such that a separate search isn't required but a custom config can be input if desired.